### PR TITLE
add FileReader Web API to js dom

### PR DESCRIPTION
**Pull request by cooldome**
FileReader is web api that is used to display content of file before uploading it.

Details of API can be found here:
   https://developer.mozilla.org/en-US/docs/Web/API/FileReader
   https://w3c.github.io/FileAPI/#dfn-filereader

All new types and procs are within `since(1,3)`

### CODE REVIEW

File: lib/js/dom.nim [hunk 1]
```diff
@@ -39,6 +39,10 @@ type
     onselect*: proc (event: Event) {.nimcall.}
     onsubmit*: proc (event: Event) {.nimcall.}
     onunload*: proc (event: Event) {.nimcall.}
+    when (NimMajor, NimMinor) >= (1, 3):
```
Comment (Clyybber) [inline lib/js/dom.nim]:
```suggestion
    since (1, 3):
```

Comment (cooldome) [reply lib/js/dom.nim]: You can't use templates insode object type definitions, only direct when statements are allowed

Comment (cooldome) [reply lib/js/dom.nim]: In the end I had to remove `when` as well. It doesn't work, semfold chokes.

File: lib/js/dom.nim [hunk 2]
```diff
@@ -1318,6 +1345,29 @@ proc offsetLeft*(e: Node): int {.importcpp: "#.offsetLeft", nodecl.}
 since (1, 3):
   func newDomParser*(): DOMParser {.importcpp: "new DOMParser()".}
     ## DOM Parser constructor.
-
   func parseFromString*(this: DOMParser; str: cstring; mimeType: cstring): Document {.importcpp.}
     ## Parse from string to `Document`.
+
+  proc newDomException*(): DomException {.importcpp: "new DomException()", constructor.}
```
Comment (juancarlospaco) [inline lib/js/dom.nim:1351]:
```suggestion
  proc newDomException*(): DomException {.importcpp: "(new DomException())", constructor.}
```

File: lib/js/dom.nim [hunk 3]
```diff
@@ -1318,6 +1345,29 @@ proc offsetLeft*(e: Node): int {.importcpp: "#.offsetLeft", nodecl.}
...
+    ## DOM Exception constructor
+  proc message*(ex: DomException): cstring {.importcpp: "#.message", nodecl.}
+    ## https://developer.mozilla.org/en-US/docs/Web/API/DOMException/message
+  proc name*(ex: DomException): cstring  {.importcpp: "#.name", nodecl.}
+    ## https://developer.mozilla.org/en-US/docs/Web/API/DOMException/name
+
+  proc newFileReader*(): FileReader {.importcpp: "new FileReader()", constructor.}
```
Comment (juancarlospaco) [inline lib/js/dom.nim:1358]:
```suggestion
  proc newFileReader*(): FileReader {.importcpp: "(new FileReader())", constructor.}
```

Comment (cooldome) [reply lib/js/dom.nim:1358]: I don't think you right, wrapping in the parenthesis is not required, have a look how all other objects in this module are implemented

File: lib/js/dom.nim (hunk 2 above)
Comment (cooldome) [reply lib/js/dom.nim:1351]: I don't think you right, wrapping in the parenthesis is not required, have a look how all other objects in this module are implemented

File: lib/js/dom.nim [hunk 4]
```diff
@@ -39,6 +39,9 @@ type
     onselect*: proc (event: Event) {.nimcall.}
     onsubmit*: proc (event: Event) {.nimcall.}
     onunload*: proc (event: Event) {.nimcall.}
+    onloadstart*: proc (event: Event) {.nimcall.}
```
Comment (Araq) [inline lib/js/dom.nim:42]: well now this can work inside a ``when``.

Comment (cooldome) [reply lib/js/dom.nim:42]: ok, let's try

**Comment by cooldome** [comment]
Ready for review

**Comment by cooldome** [comment]
tried. Ci can't pickup nim changes that quickly. reverted last change.

**Pull request merged**